### ISSUE TITLE
Collection expressions: CS0121 ambiguous call between `string` and `char[]`

### ISSUE THREAD

**Issue by jnyrup** [dotnet/roslyn#70701]
**Version Used**: 
Visual Studio Version 17.8.0 Preview 6.0

**Steps to Reproduce**:
```cs
class Class1
{
    void Foo0() => Bar(new[] { 'a' }); // calls Bar(char[] _)
    void Foo1() => Bar("a"); // Bar(string _)
    void Foo2() => Bar(['a']); // triggers CS0121

    void Bar(char[] _) { }
    void Bar(string _) { }
}
```
[SharpLab](https://sharplab.io/#v2:EYLgtghglgdgNAFxAJwK4wD4AEBMBGAWAChjcACAQWIG9iz6ysAWMgMQHt2AGACgEoyAXgB8ZAEIRkPGAFMA7gG0AumWpkA5BHVkAvnwDcdBszac8/IaIlSARBBsGj9Ex3Y4LI8ZJ4LN6pY5EDGROIUHGLNY8AMYAFpLKZAD6Amo6oSZRWHhcyam6xDrBoUA)

**Diagnostic Id**:
```
CS0121	The call is ambiguous between the following methods or properties: 'Class1.Bar(char[])' and 'Class1.Bar(string)'
```

**Expected Behavior**:
Overload resolution for `Bar` in `Foo2` should prefer `Bar(char[])` over `Bar(string)`

**Actual Behavior**:
CS0121 is triggered.

**Comment by cston** [comment]
@jnyrup, the current behavior is expected.

`['a']` is considered convertible to `string`, even though the compiler will report an error constructing the `string` instance (see [LDM-2023-10-02](https://github.com/dotnet/csharplang/blob/main/meetings/2023/LDM-2023-10-02.md#conclusions)), so the call is ambiguous.

**Comment by jnyrup** [comment]
Thanks for the reference.
Since collection expressions are not _currently_ convertible to `string` I assumed the ambiguity was a bug. 

Feel free to close this.
